src/strategies/nation3-votes-with-delegations/index.ts

```diff
@@ -0,0 +1,124 @@
+import { BigNumber, BigNumberish } from '@ethersproject/bignumber';
+import { formatUnits } from '@ethersproject/units';
+import { Multicaller } from '../../utils';
+
+export const author = 'caranell';
+export const version = '0.1.0';
+const DECIMALS = 18;
+
+const balanceAbi = [
+  'function balanceOf(address account) external view returns (uint256)'
+];
+
+const ownerAbi = ['function ownerOf(uint256 id) public view returns (address)'];
+
+const signerAbi = [
+  'function signerOf(uint256 id) external view  returns (address)'
+];
+
+const lastTokenIdAbi = ['function getNextId() external view returns (uint256)'];
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+): Promise<Record<string, number>> {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  const erc721OwnerCaller = new Multicaller(network, provider, ownerAbi, {
+    blockTag
+  });
+  const erc721SignerCaller = new Multicaller(network, provider, signerAbi, {
+    blockTag
+  });
+  const erc721LastTokenIdCaller = new Multicaller(
+    network,
+    provider,
+    lastTokenIdAbi,
+    { blockTag }
+  );
+
+  const erc20BalanceCaller = new Multicaller(network, provider, balanceAbi, {
+    blockTag
+  });
+
+  erc721LastTokenIdCaller.call('lastTokenId', options.erc721, 'getNextId');
+
+  addresses.forEach((address) => {
+    erc20BalanceCaller.call(address, options.erc20, 'balanceOf', [address]);
+  });
+
+  const [erc20Balances, lastIndex]: [
+    Record<string, BigNumberish>,
+    Record<string, BigNumberish>
+  ] = await Promise.all([
+    erc20BalanceCaller.execute(),
+    erc721LastTokenIdCaller.execute()
+  ]);
+
+  const lastTokenId = BigNumber.from(lastIndex.lastTokenId).toNumber();
+
+  for (let i = 0; i < lastTokenId; i++) {
+    erc721SignerCaller.call(i, options.erc721, 'signerOf', [i]);
+    erc721OwnerCaller.call(i, options.erc721, 'ownerOf', [i]);
+  }
```

Comment:
Hey, @Caranell Everything looks good to me :) except that I feel someone might pass their custom contract for this and lastTokenId could be a huge number and it could eat up all memory. of course, it is a rare case but would be great if we can have a limit if it crosses that limit we can `throw` an error. LMK what you think

Reply:
Hmm, yea no idea too 😄 I am starting to worry that this may not scale well if there are more tokens in your passport contract, there is no other way to find delegates than to get all owners and signers for each token.? maybe a subgraph or something?

Reply:
Hi @ChaituVR. I discussed the problem with the nation3 team and we don't expect the number of 'passport tokens' to exceed even a thousand in the near future. So I suggest leaving the current behavior as it is, and if the number of tokens significantly increases, we'll use subgraphs. What do you think?